### ISSUE TITLE
Missing attributes information on `setter` of a property (in `FSharpMemberOrFunctionOrValue` type)

### ISSUE THREAD

**Issue by MangelMaxime** [dotnet/fsharp#18157]
Hello,

I am not sure how to report this bug because this is related to the internals of F# Compiler itself, and happening inside of Fable transformation.

In Fable, we want to erase types and properties decorated with `[<Erase>]`.

For example, this code should emit nothing.

```fs
[<Erase>]
type internal LanguageInjectionAttribute() =
    inherit Attribute()
    
    [<Erase>]
    member val Prefix = "" with get, set
```

However, currently it does not seems possible to do it for the `setter` because it is missing the `[<Erase>]` attribute information:

![Image](https://github.com/user-attachments/assets/5acc1adb-d3c6-4718-bbf2-e17e1f0604ed)

Please tell me, if I can do anything to reword my issue or help you identify the cause of this problem.

**Comment by vzarytovskii** [comment]
Yeah, I don't think attributes are universally propagated to property methods. One of the reasons is that targets on attributes can be incorrect for methods. Which is fine from the runtime perspective, but semantically incorrect.

**Comment by T-Gro** [comment]
You should be able to find the attribute on the symbol for the whole Property (and not just getter/setter).
Is it there?

**Comment by MangelMaxime** [comment]
@T-Gro What do you mean by Symbol in this context?

I tried to look but `FSharpMemberOrFunctionOrValue` does not seems to have a `Symbol` property for access.

However, I found that there is a `DeclaringEntity` which seems to have the `Attributes` information, would this be a good place to look for `Attributes` if we detect that the `FSharpMemberOrFunctionOrValue` is a `setter`?

![Image](https://github.com/user-attachments/assets/63bf3849-1259-4969-bbfd-3d306e08b130)

**Comment by MangelMaxime** [comment]
> `.DeclaringEntity.Attributes` gets attributes of the type, not of the property - I assume this is not what you want.

This explains why I had a strange result when removing `[<Erase>]` from the top level entity.

> `.DeclaringEntity.MembersFunctionsAndValues` should yield a collection of members, one of which will be a `P` with `PropInfo` (i.e. the full property, not just the `set_...` method. Can you locate the attribute like that?
> 
> (locate the one you want via `.LogicalName` and `.IsProperty`)

This seems possible indeed, it just make the code more verbose compared to the other case where `Attributes` is propagated.

```fs
let isErasedPropertySetter (memb : FSharpMemberOrFunctionOrValue) =
    if memb.IsPropertySetterMethod then
        match memb.DeclaringEntity with
        | Some ent ->
            // Remove the "set_" prefix
            let logicalName = memb.LogicalName.Substring(4)

            ent.MembersFunctionsAndValues
            |> Seq.tryFind(fun (m : FSharpMemberOrFunctionOrValue) ->
                m.LogicalName = logicalName && hasAttrib Atts.erase m.Attributes
            )
            |> Option.isSome
        | _ -> false
    else
        false
``` 

Depending on if the attributes should be propagated to the `setter`, this can be a workaround or the final solution.

Thank you for the guidance.

**Comment by T-Gro** [comment]
I think automatic propagation might be a breaking change, since AttributeTargets.Property and AttributeTargets.Method are two distinct options.

https://learn.microsoft.com/en-us/dotnet/api/system.attributetargets?view=net-8.0

(i.e. it might lead to an error saying that attribute intended for properties only appeared on a method)

**Comment by edgarfgp** [comment]
> I think automatic propagation might be a breaking change, since AttributeTargets.Property and AttributeTargets.Method are two distinct options.
> 
> https://learn.microsoft.com/en-us/dotnet/api/system.attributetargets?view=net-8.0
> 
> (i.e. it might lead to an error saying that attribute intended for properties only appeared on a method)

I think the attribute not appearing on members here is a bug(regardless the targets it should be using). It is will be a breaking change in the sense that user defined attributes will have to adjust the targets but like we have done in F#9 for `AttributeTarget` changes is a small price to pay to make sure we address the still long standing issues with attributes in F#

**Comment by edgarfgp** [comment]
I’d be happy to investigate and potentially fix this if we want to address this

**Comment by T-Gro** [comment]
They really should not be mixed up.
Properties and methods are different elements in the IL, and they can each carry different attributes.

See e.g. here:
https://sharplab.io/#v2:EYLgtghglgdgNAFxFANgHwAICYCMBYAKEIG0BBBBAJymAFcEBTAVQGcIBzBgCnKpvoYAVCJU4IWAOgAKlAPYAHBpQQBPAJQBdQhgDMAAmx6Asit6Uz/RnpB6LdRoQDehAL6F3BXQax6AwnucCPWDiEzM7AS0g4K8MHAAGAH5jFRkFJVUAsQBuFgYEbLcCFyA


The usability aspect which @MangelMaxime mentions is IMO the right target here - the exposed Symbols API should make it easier to traverse the data from method to property or vice versa (without having to do the string slicing which is visible in the snippet)
